src/storage/copier/npy_reader.cpp

```diff
@@ -224,10 +224,12 @@ std::shared_ptr<arrow::DataType> NpyReader::getArrowType() const {
 std::shared_ptr<arrow::RecordBatch> NpyReader::readBlock(common::block_idx_t blockIdx) const {
```

Comment:
The returned result is incorrect for fixed-sized list. Arrow's fixed-sized list requires special construction. It should be something similar to this:
```
    auto buffer = std::make_shared<arrow::Buffer>(rowPointer, CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY);
    int64_t length = std::min(CopyConstants::NUM_ROWS_PER_BLOCK_FOR_NPY, getNumRows() - rowNumber);
    std::shared_ptr<arrow::Field> field;
    std::shared_ptr<arrow::Array> arr;
    if (getNumDimensions() > 1) {
        auto elementField = std::make_shared<arrow::Field>(defaultFieldName, arrowType);
        auto fixedListArrowType = arrow::fixed_size_list(elementField, (int32_t)length);
        field = std::make_shared<arrow::Field>(defaultFieldName, fixedListArrowType);
        auto valuesArr = std::make_shared<arrow::PrimitiveArray>(arrowType, length * getNumElementsPerRow(), buffer);
        arr = arrow::FixedSizeListArray::FromArrays(valuesArr, (int32_t)getNumElementsPerRow()).ValueOrDie();
    } else {
        field = std::make_shared<arrow::Field>(defaultFieldName, arrowType);
        arr = std::make_shared<arrow::PrimitiveArray>(arrowType, length, buffer);
    }
```